EssentialFeed/EssentialFeed/Feed Feature/FeedCommentsLoader.swift

```diff
@@ -0,0 +1,17 @@
+//
+//  Created by Azamat Valitov on 13.03.2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+
+public protocol FeedCommentsLoaderTask {
+	func cancel()
+}
+
+public protocol FeedCommentsLoader {
+	typealias Result = Swift.Result<[FeedComment], Error>
+	
+	@discardableResult
+	func load(url: URL, completion: @escaping (Result) -> Void) -> FeedCommentsLoaderTask
```

Comment:
The task return shouldn't be discardable. It should be held by clients - never ignored.

Moreover, `URL` is an implementation detail of the `Remote` loader. Not *every* implementation should need a URL.

The protocol should not expose implementation details. Otherwise, it's not a good abstraction.

The URL should be injected into the concrete Remote loader implementation via initializer injection instead.

```suggestion
	func load(completion: @escaping (Result) -> Void) -> FeedCommentsLoaderTask
```